v2/internal/controllers/owner_arm_id_negative_test.go

```diff
@@ -53,7 +53,7 @@ func Test_OwnerIsARMIDOfWrongType_Rejected(t *testing.T) {
 	tc.Expect(err).To(MatchError(ContainSubstring("expected owner ARM ID to be for a resource group, but was \"Microsoft.Compute/virtualMachines\"")))
 }
 
-func Test_OwnerIsARMIDFromDifferentSubscription_ResourceFails(t *testing.T) {
+func Test_OwnerIsNotARMIDFromDifferentSubscription_ResourceFails(t *testing.T) {
```

Comment:
minor: naming might be better as 
```suggestion
func Test_OwnerIsKubernetesResourceFromDifferentSubscription_ResourceFails(t *testing.T) {
```

(saying what the owner is, rather than what it is not)